test/test_torch.py

```diff
@@ -953,6 +953,27 @@ def test_complex_half_experimental_warning(self, device):
             # t + 1 allocates a new tensor for result using empty
             t + 1
 
+    @onlyCUDA
+    def test_dtypetensor_warnings(self, device):
+        msg = 'The torch.cuda.*DtypeTensor constructors are no longer recommended'
+        with self.assertWarnsOnceRegex(UserWarning, msg):
+            t = torch.cuda.FloatTensor([0])
+
+        with self.assertWarnsOnceRegex(UserWarning, msg):
+            t = torch.cuda.DoubleTensor([0])
+
+    def test_set_default_tensor_type_warnings(self, device):
+        msg = '.*is deprecated as of PyTorch 2.1, please use torch.set_default_dtype().*'
+        default_type = torch.tensor([]).type()
+        with self.assertWarnsOnceRegex(UserWarning, msg):
+            torch.set_default_tensor_type(torch.FloatTensor)
+
+        if torch.cuda.is_available():
+            with self.assertWarnsOnceRegex(UserWarning, msg):
+                torch.set_default_tensor_type(torch.cuda.FloatTensor)
+
+        torch.set_default_tensor_type(default_type)
```

Comment:
Please use a `try/finally` block to make sure you always restore the state here even if there was an error during this test.

Reply:
Good advice.